android/src/oldarch/io/customer/reactnative/sdk/messaginginapp/NativeMessagingInAppModule.kt

```diff
@@ -0,0 +1,57 @@
+package io.customer.reactnative.sdk.messaginginapp
+
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.modules.core.DeviceEventManagerModule
+
+/**
+ * React Native module implementation for Customer.io In-App Messaging Native
+ * SDK using old architecture.
+ */
+class NativeMessagingInAppModule(
+    private val reactContext: ReactApplicationContext,
+) : ReactContextBaseJavaModule(reactContext) {
+    override fun getName(): String = NativeMessagingInAppModuleImpl.NAME
+
+    private var listenerCount = 0
+    private val inAppEventListener: ReactInAppEventListener
+        get() = NativeMessagingInAppModuleImpl.inAppEventListener
+
+    override fun initialize() {
+        super.initialize()
+        inAppEventListener.setEventEmitter { data ->
+            if (listenerCount <= 0) return@setEventEmitter
+
+            reactContext
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
+                .emit("InAppEventListener", data)
```

Comment:
should we be emitting it from ui thread? to avoid “calling JS from non-JS thread” since original would be from main thread?

Reply:
[React docs](https://reactnative.dev/docs/legacy/native-modules-android#sending-events-to-javascript) don't really mention any need for it, so probably it handles by itself. But I'll double check and update if needed